### PR TITLE
yab bump version

### PR DESCRIPTION
**Pull request by bbjimmy**

### CODE REVIEW

File: dev-lang/yab/yab-1.7.5.6.recipe [hunk 1]
```diff
@@ -4,12 +4,12 @@ language, with special commands designed for Haiku."
...
 	2006-2009 Jan Bungeroth (yab improvements)
-	2013-2017 Jim Saxton (yab improvements)"
+	2013-2018 Jim Saxton (yab improvements)"
 LICENSE="Artistic"
-REVISION="2"
-SOURCE_URI="https://github.com/bbjimmy/YAB/archive/1.7.5.5.tar.gz"
-CHECKSUM_SHA256="ea80d21487c472b7ae11eb4f45bb6dc3468b0e68490a7a1ea2d5c44378dc1853"
-SOURCE_DIR="YAB-1.7.5.5"
+REVISION="1"
+SOURCE_URI="https://github.com/bbjimmy/YAB/archive/1.7.5.6.tar.gz"
```
Comment (extrowerk) [inline dev-lang/yab/yab-1.7.5.6.recipe]: -SOURCE_URI="https://github.com/bbjimmy/YAB/archive/1.7.5.6.tar.gz"
+SOURCE_URI="https://github.com/bbjimmy/YAB/archive/$portVersion.tar.gz"

File: dev-lang/yab/yab-1.7.5.6.recipe [hunk 2]
```diff
@@ -4,12 +4,12 @@ language, with special commands designed for Haiku."
...
+CHECKSUM_SHA256="f579c532547226dec62eb6c639f6e53ee3e82282d62077e864a3bc3889d28c89"
+SOURCE_DIR="YAB-1.7.5.6"
```
Comment (extrowerk) [inline dev-lang/yab/yab-1.7.5.6.recipe]: -SOURCE_DIR="YAB-1.7.5.6"
+SOURCE_DIR="YAB-$portVersion"

**Comment by extrowerk** [comment]
In BUILD():
```
	mkdir -p tmp
	cp -r src/* tmp
	cd tmp
```
not required.

**Comment by extrowerk** [comment]
In INSTALL():
```
	mkdir -p $binDir
	mkdir -p $libDir
	mkdir -p $appsDir/yab-IDE
```
could be just:
`	mkdir -p $binDir $libDir $appsDir/yab-IDE`

File: dev-lang/yab/yab-1.7.5.6.recipe (hunk 2 above)
Comment (fbrosson) [reply dev-lang/yab/yab-1.7.5.6.recipe]: While at it, could you also add a `SOUCE_FILENAME`?
```diff
-SOURCE_URI="https://github.com/bbjimmy/YAB/archive/1.7.5.6.tar.gz"
+SOURCE_URI="https://github.com/bbjimmy/YAB/archive/$portVersion.tar.gz"
 CHECKSUM_SHA256="f579c532547226dec62eb6c639f6e53ee3e82282d62077e864a3bc3889d28c89"
+SOURCE_FILENAME="YAB-$portVersion.tar.gz"
-SOURCE_DIR="YAB-1.7.5.6"
+SOURCE_DIR="YAB-$portVersion"
```
Thanks!

Useful links:
* [HaikuPorter-Guidelines#archive-name](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#archive-name)
* [HaikuPorter-Guidelines#ordering](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering)

**Comment by bbjimmy** [comment]
There is no guarantee that the next release will be named so that
$portVersion can be used. Sometimes there is a change that needs to be
added after a release is made and the final release ends up with a
different name.

**Comment by bbjimmy** [comment]
Be my guest. I just made a working recipe based on the one in haikuports.

**Comment by bbjimmy** [comment]
SOURCE_URI="https://github.com/bbjimmy/YAB/archive/$portVersiona.tar.gz"
translates to 
https://github.com/bbjimmy/YAB/archive/.tar.gz 

instead of

https://github.com/bbjimmy/YAB/archive/1.7.5.6a.tar.gz
SOURCE_FILENAME="YAB-$portVersiona.tar.gz"

translates to 

YAB-.tar.gz

instead of 

YAB-1.7.5.6a.tar.gz

SOURCE_DIR="YAB-$portVersiona"
 translates to
 YAB- 
instead of

 YAB-1.7.5.6a

IE $portVersion dose not work here.

**Comment by KapiX** [comment]
Use `${portVersion}a`.

**Comment by fbrosson** [comment]
```diff
-SOURCE_FILENAME="YAB-$portVersiona.tar.gz"
+SOURCE_FILENAME="YAB-${portVersion}a.tar.gz"
```

**Comment by fbrosson** [comment]
Hmm, you might wish to rename the recipe to `yab-1.7.5.6a.recipe` (and use `$portVersion` instead of `${portVersion}a`).

**Comment by fbrosson** [comment]
This is strange. Are you sure you did not add or eat some character somewhere?

Anyway, if you rename your recipe to `yab-1.7.5.6a.recipe` then you will be able to use:
```bash
SOURCE_URI="https://github.com/bbjimmy/YAB/archive/$portVersion.tar.gz"
CHECKSUM_SHA256="f26ac72c852047c15a552eeff1a52388f75b244d9173883f3183e52d9e5795c0"
SOURCE_FILENAME="YAB-$portVersion.tar.gz"
SOURCE_DIR="YAB-$portVersion"
```

**Comment by bbjimmy** [comment]
OK. I had to use Windows, Haiku has no browswer that works completely on github.  This allowed me to delete and re-create the release so now it uses $portVersion

**Comment by bbjimmy** [comment]
All the issues have been resolved.

This should be merged.

-- 
bbjimmy

http://www.coquillemartialarts.com
http://www.fatelk.com
http://yab.orgfree.com

**Comment by fbrosson** [comment]
While I was trying to figure out wether or not YAB could be changed to work with a different install layout (i.e. the app in `$appsDIr` with no subdirectries in there, and the data files in `$data/yab/`) I noticed that there's something in `YAB-1.7.5.6/src/Makefile` that could be changed a little:
```diff
 ##
 ## Find haiku systenm archetecture from the package file name
 ##
-SYSTEMARCH:= $(shell catattr SYS:PACKAGE_FILE /boot/system/kernel_x86*|cut --fields=4 -d-|cut -d. --fields 1)
+SYSTEMARCH:= $(shell getarch --primary)
```
Of course, this is a neutral change, so you might wish to adopt it for the next release only.

BTW, because of the current install layout I won't merge this PR.
That said, I don't object a merge since most contributors are OK with this layout.

**Comment by bbjimmy** [comment]
yab-IDE uses these files to build binary files out of scripts. It
needs the currently installed yab files to do this.

**Comment by alaviss** [comment]
@fbrosson 

In case you're as confused as I did about why the source was there, you can take a look at my conversation with @bbjimmy on [IRC][0], starts at `[07:01:23]`

**TL;DR**: If the package builds, I will merge, then we will discuss whether if we should split the source code to a separate package in an issue.

[0]: https://echelog.com/logs/browse/haiku/1525730400
